Source/Lib/Codec/EbPacketizationProcess.c

```diff
@@ -569,7 +569,7 @@ void* PacketizationKernel(void *inputPtr)
         // Encode slice header and write it into the bitstream.
         packetizationQp = pictureControlSetPtr->pictureQp;
      
-        if(sequenceControlSetPtr->staticConfig.accessUnitDelimiter && (pictureControlSetPtr->pictureNumber > 0)) 
+        if (sequenceControlSetPtr->staticConfig.accessUnitDelimiter && !(sequenceControlSetPtr->staticConfig.codeVpsSpsPps && toInsertHeaders))
```

Comment:
Hi @phamd ,

I don't think it's a reasonable logic to encode AUD, as it's just a condition mutually exclusive with the previous one in the same PacketizationKernel() function. Actually EbH265EncStreamHeader() would also encode AUD once it's invoked by application. Please reconsider your change. Thanks!

Reply:
With `rateControlMode=1` and `intraRefreshType=0`, the current condition `(pictureControlSetPtr->pictureNumber > 0)` isn't mutually exclusive with `toInsertHeaders`; I'm seeing this state with 2 AUD in the output ES. But it could be the case that there's a different bug in this block:

```c
        toInsertHeaders = EB_FALSE;
        if (pictureControlSetPtr->pictureNumber == 0) {
            toInsertHeaders = EB_TRUE;
        } else if ((pictureControlSetPtr->sliceType == EB_I_PICTURE) &&
                   (sequenceControlSetPtr->intraRefreshType >= IDR_REFRESH)) {
            if (sequenceControlSetPtr->staticConfig.rateControlMode) {
                EB_U32 idrCount = pictureControlSetPtr->pictureNumber /
                                  (sequenceControlSetPtr->intraPeriodLength + 1);
                if ((idrCount % (sequenceControlSetPtr->intraRefreshType + 1)) == 0)
                    toInsertHeaders = EB_TRUE;
            } else if (sequenceControlSetPtr->intraRefreshType == IDR_REFRESH) {
                toInsertHeaders = EB_TRUE;
            }
        }
```

Reply:
Hi @phamd ,

I was saying, your change is just alternative condition compared with Line 260 in this file, but maybe it's not a reasonable logic to trigger AUD encoded. Thanks!